### PR TITLE
markdown: add GOVUKFrontendExtension to apply govuk-frontend classes to generated elements

### PR DESCRIPTION
**Pull request by risicle**
https://trello.com/c/scaNu2Aq

What do people think? To me, doing this seems less scary than keeping half of our content fields as fully-escaped html.

~Haven't tried this in a frontend yet.~

### CODE REVIEW

File: dmcontent/markdown.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    Markdown extension adding govuk-frontend classes to generated paragraphs and lists (note - will not affect
+    manual html)
+    """
+    def extendMarkdown(self, md, md_globals):
+        # NOTE the interface for registering extensions is much improved after Markdown 3.0.0. TODO upgrade.
+        md.registerExtension(self)
+        self.processor = _ClassAddingTreeprocessor({
+            "p": "govuk-body",
+            "ul": "govuk-list govuk-list--bullet",
+            "ol": "govuk-list govuk-list--number",
```
Comment (katstevens) [inline dmcontent/markdown.py]: Could we pull this out separately so it's super easy to append additional classes to later if needed?

File: dmcontent/markdown.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        # mutate each affected element in-place
+        if element.tag in self._mapping:
+            orig_class = element.attrib.get("class", "")
+            element.set("class", (orig_class + " " if orig_class else "") + self._mapping[element.tag])
+
+
+class GovUkFrontendExtension(Extension):
+    """
+    Markdown extension adding govuk-frontend classes to generated paragraphs and lists (note - will not affect
+    manual html)
```
Comment (katstevens) [inline dmcontent/markdown.py]: What happens here with escaping? Could we add a test?

File: dmcontent/markdown.py (hunk 1 above)
Comment (risicle) [reply dmcontent/markdown.py]: Yes, there's even a mechanism for doing that built into the extension business but I decided not to wrestle with that now seeing as I didn't even know where the config should be kept.

File: dmcontent/markdown.py (hunk 2 above)
Comment (risicle) [reply dmcontent/markdown.py]: Oh I thought I included an ampersand in the test text but I didn't. I'll do that.

Comment (risicle) [reply dmcontent/markdown.py]: Done

File: dmcontent/markdown.py (hunk 1 above)
Comment (risicle) [reply dmcontent/markdown.py]: Done

**Comment by katstevens** [comment]
Cool 👍

**Comment by risicle** [comment]
Does this look ok to people? It's actually only a little bit wider spaced than it was before
Spectacle h32487

**Comment by risicle** [comment]
Before
Spectacle n32487

**Comment by risicle** [comment]
Though it does fix the wrong-sized list text font problem with narrow screens
Spectacle k32487

**Comment by risicle** [comment]
Whereas without the new classes...
Spectacle T32487

**Comment by katstevens** [comment]
Yep, spacing looks fine to me (sizing fix obviously a win as well).

File: dmcontent/markdown.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        for child in element:
+            self._run(child)
+
+        # mutate each affected element in-place
+        if element.tag in self._mapping:
+            orig_class = element.attrib.get("class", "")
+            element.set("class", (orig_class + " " if orig_class else "") + self._mapping[element.tag])
+
+
+class GovUkFrontendExtension(Extension):
```
Comment (lfdebrux) [inline dmcontent/markdown.py]: This should probably be named `GOVUKFrontendExtension` to be consistent with [govuk-frontend](https://github.com/alphagov/govuk-frontend/blob/899264cd89559d31f0a790682254d94ecf66418b/package/govuk/common.js)

Comment (risicle) [reply dmcontent/markdown.py]: :+1:

**Comment by katstevens** [comment]
Re-running the Py3.8 Travis check, which stalled trying to build its environment.

**Pull request merged**